kafka-ui-api/src/main/java/com/provectus/kafka/ui/service/ConsumingService.java

```diff
@@ -116,4 +130,36 @@ private boolean filterTopicMessage(TopicMessageEvent message, String query) {
         || (!StringUtils.isEmpty(msg.getContent()) && msg.getContent().contains(query));
   }
 
+  private void evalJsFilterFn(String jsFilterFn) {
+    try {
+      jsEngine.eval(jsFilterFn);
```

Comment:
is it right that we need this only for function registration? what if we already registered and using it and another search request appears?

Reply:
Yes, we declare our filter function here. It's a good point, that parallel requests could have collision. I'll try to fix it

Reply:
Could you check it again cc @iliax? I'm still declaring fn, but with uniq names and remove fn's after terminating. Maybe we can discuss it, if you have ideas?